internal/controllers/rollout/controller.go

```diff
@@ -25,114 +22,81 @@ type controller struct {
 	cooldown time.Duration
 }
 
-// NewController re-synthesizes compositions when their synthesizer has changed while honoring a cooldown period.
 func NewController(mgr ctrl.Manager, cooldown time.Duration) error {
 	c := &controller{
 		client:   mgr.GetClient(),
 		cooldown: cooldown,
 	}
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&apiv1.Synthesizer{}).
-		Watches(&apiv1.Composition{}, newCompositionHandler()).
-		// TODO: Filter some events?
-		WithLogConstructor(manager.NewLogConstructor(mgr, "synthesizerRolloutController")).
+		Named("rolloutController").
+		Watches(&apiv1.Composition{}, manager.SingleEventHandler()).
```

Comment:
We use `manager.SingleEventHandler` so that `RequeueAfter` results will be honored across compositions. Otherwise there's a risk of dispatching another synthesis before the previously dispatched one hits the informer.